### PR TITLE
JTD-020 add migration to add primary key to client table

### PR DESCRIPTION
**Pull request by dana-kainos**

### CODE REVIEW

File: src/main/resources/db/migration/V4__add_primary_key_to_client_table.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE client MODIFY clientId int NOT NULL PRIMARY KEY;
```
Comment (lukemKainos) [inline src/main/resources/db/migration/V4__add_primary_key_to_client_table.sql]: can you check the syntax on this please

**Comment by dana-kainos** [comment]
turns out it didnt even work in the first place

**Comment by dana-kainos** [comment]
has been tested now and it actually works

File: src/main/resources/db/migration/V1__create_client_table.sql [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-    postcode nvarchar(10),
-    country char(2) NOT NULL
+    addressLine1 varchar(50) NOT NULL,
+    addressLine2 varchar(50),
+    town varchar(50),
+    city varchar(50),
+    postcode varchar(10),
+    country char(2) NOT NULL,
     phone varchar(15)
 )
```
Comment (lukemKainos) [inline src/main/resources/db/migration/V1__create_client_table.sql]: Please correct the name of this PR

File: src/main/resources/db/migration/V1__create_client_table.sql [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-    city nvarchar(50),
-    postcode nvarchar(10),
-    country char(2) NOT NULL
+    addressLine1 varchar(50) NOT NULL,
+    addressLine2 varchar(50),
+    town varchar(50),
+    city varchar(50),
+    postcode varchar(10),
+    country char(2) NOT NULL,
     phone varchar(15)
```
Comment (lukemKainos) [inline src/main/resources/db/migration/V1__create_client_table.sql]: if we are doing nvarchar for the rest why not here? also can this be null? do we have a form of contact?

File: src/main/resources/db/migration/V1__create_client_table.sql [hunk 4]
```diff
@@ -1,11 +1,11 @@
 CREATE TABLE `client` (
-    clientId int AUTO_INCREMENT NOT NULL,
+    clientId int AUTO_INCREMENT NOT NULL PRIMARY KEY,
     name nvarchar(50) NOT NULL,
-    addressLine1 nvarchar(50) NOT NULL,
-    addressLine2 nvarchar(50),
-    town nvarchar(50),
-    city nvarchar(50),
-    postcode nvarchar(10),
-    country char(2) NOT NULL
```
Comment (lukemKainos) [inline src/main/resources/db/migration/V1__create_client_table.sql:9]: why is this only 2 chars

**Review by lukemKainos — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/V1__create_client_table.sql (hunk 4 above)
Comment (dana-kainos) [reply src/main/resources/db/migration/V1__create_client_table.sql:9]: ISO3166 country code as specified in pr JTD-001

**Comment by lukemKainos** [comment]
> turns out it didnt even work in the first place

please clean up the comments on this pr. Unnecessary comments.

File: src/main/resources/db/migration/V1__create_client_table.sql (hunk 3 above)
Comment (dana-kainos) [reply src/main/resources/db/migration/V1__create_client_table.sql]: can what be null? is phone number not a form of contact?

File: src/main/resources/db/migration/V1__create_client_table.sql [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
-    phone varchar(15)
+    addressLine1 varchar(50) NOT NULL,
+    addressLine2 varchar(50),
+    town varchar(50),
+    city varchar(50),
+    postcode varchar(10),
+    country char(2) NOT NULL,
+    phone varchar(15),
+    email varchar(50)
```
Comment (lukemKainos) [inline src/main/resources/db/migration/V1__create_client_table.sql]: neither of these are mandatory, we need a way to contact

**Review by lukemKainos — APPROVED** [review]

**Review by dylanmckk — APPROVED** [review]

**Pull request merged**